src/core/modules/Nebula.cpp

```diff
@@ -811,13 +808,64 @@ void Nebula::drawHints(StelPainter& sPainter, float maxMagHints, StelCore *core)
 		case NebEn:
 		case NebSNC:
 		case NebSNRC:
-			if (finalSize > 75.f)
-				Nebula::texDiffuseNebulaXLarge->bind();
-			else if (finalSize > 35.f)
-				Nebula::texDiffuseNebulaLarge->bind();
-			else
-				Nebula::texDiffuseNebula->bind();
-			break;
+		{
+			// Take into account device pixel density and global scale ratio, as we are drawing 2D stuff.
+			const auto pixelRatio = sPainter.getProjector()->getDevicePixelsPerPixel();
+			const auto scale = pixelRatio * StelApp::getInstance().getGlobalScalingRatio();
+			finalSize *= scale;
+
+			const float x = XY[0];
+			const float y = XY[1];
+			const float roundRadius = 0.35 * finalSize;
+			const int numPointsInArc = std::lround(std::clamp(5*finalSize/35, 5.f, 16.f));
+			std::vector<float> vertexData;
+			vertexData.reserve(numPointsInArc*2*4);
+			const float leftOuterX = x - finalSize;
+			const float leftInnerX = leftOuterX + roundRadius;
+			const float bottomOuterY = y - finalSize;
+			const float bottomInnerY = bottomOuterY + roundRadius;
+			const float rightOuterX = x + finalSize;
+			const float rightInnerX = rightOuterX - roundRadius;
+			const float topOuterY = y + finalSize;
+			const float topInnerY = topOuterY - roundRadius;
+			using std::sin;
+			using std::cos;
+			for(int n = 0; n < numPointsInArc; ++n)
```

Comment:
Not sure: I see four loops which all use the same angle and sin/cos values. Would it not be advisable (esp. for lesser systems) to create those values only once, cache them in arrays and re-use in the 4 loops?

Reply:
Maybe, but do you have any evidence that this is a critical place to add more code for this caching? It'll obviously make the code less straightforward to read.

Reply:
OK, using a function from StelUtils seems to be an acceptable compromise between efficiency and readability. So I've pushed this.

Reply:
But don't you now compute values around a full circle and use 1/4 of it? ComputeCosSinRhoZone computes the necessary parts of a semicircle, this could avoid some wasted cycles. Admittedly, the StelUtils functions themselves are efficiently avoiding the sin/cos by using the addition theorem, so it may not matter much. 

I think the call would be 
StelUtils::ComputeCosSinRhoZone(M_PI/(2.f*(numPointsInArc-1)), (numPointsInArc-1), M_PIf/2.f);

but I could be 1 off. 

I can test versions on a RPi3 with many nebulae and see if that matters anywhere. I should test on an RPi3 in any case for fps differences.